### DISCUSSION TITLE
know if a file is in your public folder

### DISCUSSION THREAD

**Discussion by jphyqr** [vercel/next.js#24359] [Help]
Hello, I want to know at run time if there is a specific file in my public folder, and if so , show a <Image> if not, show a different component.

example:   `  checkForFile(`${var1}.png`) ? <Image src = {`${var1}.png}/> :  OtherComponent`

Any ideas hows?

**Comment by akellbl4** [comment]
Hi
You can do something like this:
```js
function Component({ filename }) {
  const [isExist, setExist] = useState(false)
  
  useEffect(() => {
    fetch(`/${filename}.png`, { method: 'HEAD' }).then((res) => {
      if (res.status === 200 || res.status === 204) {
        setExist(true)
      }
    })
  }, [])
  
  if (!isExist) {
    return <OtherComponent/>
  }
  
  return <Image src = {`/${filename}.png`}/>
}
```

I use `HEAD` because it's just check if file exist and doesn't load it.

**Reply by jphyqr** [reply]
ah yes, this is what i was sort of thinking, I just am not good witgh the fetch api haha.  Something to learn more about today :)
